src/passes/Metrics.cpp

```diff
@@ -62,16 +61,19 @@ struct Metrics
     counts["[tags]"] = imports.getNumDefinedTags();
     counts["[exports]"] = module->exports.size();
     counts["[tables]"] = imports.getNumDefinedTables();
-    // add memory and table
-    if (module->memory.exists) {
-      Index size = 0;
-      for (auto& segment : module->memory.segments) {
-        size += segment.data.size();
-      }
-      counts["[memory-data]"] = size;
-    }
 
+    // add memory
+    walkMemory(&module->memory);
+    for (auto& segment : module->dataSegments) {
+      walkDataSegment(segment.get());
+    }
     Index size = 0;
+    ModuleUtils::iterActiveDataSegments(
+      *module, [&](DataSegment* segment) { size += segment->data.size(); });
+    counts["[memory-data]"] = size;
```

Comment:
Looks like the table code below does
```cpp
    if (!module->tables.empty()) {
      counts["[table-data]"] = size;
    }
```
Doing the same here makes sense I think. That would also reduce the size of the test diff I believe.

Reply:
Good call out! The metrics related tests were changed because memory-data was showing with a 0 and will revert back 👍🏽